src/components/Explore/TokenTable.tsx

```diff
@@ -38,11 +44,47 @@ const LOADING_ROWS = Array(10)
     return <LoadingRow key={`${index}`} />
   })
 
+function useFilteredTokens(topTokenAddresses: string[]) {
+  const filterString = useAtomValue(filterStringAtom)
+  const favoriteTokens = useAtomValue(favoritesAtom)
+  const showFavorites = useAtomValue(showFavoritesAtom)
+  const shownTokens = showFavorites ? favoriteTokens : topTokenAddresses
+  const allTokens = useAllTokens()
+
+  return useMemo(() => {
+    const filteredTokens = shownTokens.filter((tokenAddress) => {
+      const token = allTokens[tokenAddress]
+      const tokenName = token?.name ?? ''
+      const tokenSymbol = token?.symbol ?? ''
+
+      if (!filterString) {
+        return true
+      }
+      const lowercaseFilterString = filterString.toLowerCase()
+      const addressIncludesFilterString = tokenAddress.toLowerCase().includes(lowercaseFilterString)
+      const nameIncludesFilterString = tokenName.toLowerCase().includes(lowercaseFilterString)
+      const symbolIncludesFilterString = tokenSymbol.toLowerCase().includes(lowercaseFilterString)
+      return nameIncludesFilterString || symbolIncludesFilterString || addressIncludesFilterString
+    })
+    return filteredTokens
+  }, [allTokens, shownTokens, filterString])
+}
+
+function NoTokensState(message: ReactNode, timePeriod: TimePeriod) {
```

Comment:
this should be a component

Reply:
but it uses all the same components as the table grid (since it still maintains the header and overall grid container). so I would need to export those styles which seems slightly redundant

Reply:
like this is technical the table component, just a different token display (instead of token rows, it shows none)

Reply:
I just mean you should do `NoTokensState message={} timeperiod={}`

Reply:
OH meme my b